src/bsoncxx/decimal128.hpp

```diff
@@ -0,0 +1,100 @@
+// Copyright 2014 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include <bsoncxx/stdx/string_view.hpp>
+
+#include <bsoncxx/config/prelude.hpp>
+
+namespace bsoncxx {
+BSONCXX_INLINE_NAMESPACE_BEGIN
+
+///
+/// Represents an IEEE 754-2008 Decimal128 value in a platform-independent way.
+///
+class BSONCXX_API decimal128 {
+   public:
+    ///
+    /// Constructs a Decimal128 value representing zero.
+    ///
+    decimal128() : _high(0), _low(0){};
+
+    ///
+    /// Constructs a Decimal128 from high and low 64-bit big-endian parts.
+    ///
+    /// @param high
+    ///   High bits
```

Comment:
We have no consistent style on this.  The way it renders, it might make more sense to write it as a sentence starting with the param name.  I'll try that and then putting periods at the end is more sensible.